src/Ionide.Analyzers/Suggestion/UnnamedDiscriminatedUnionFieldAnalyzer.fs

```diff
@@ -0,0 +1,43 @@
+module Ionide.Analyzers.Suggestion.UnnamedDiscriminatedUnionFieldAnalyzer
+
+open FSharp.Analyzers.SDK
+open FSharp.Analyzers.SDK.ASTCollecting
+open FSharp.Compiler.Syntax
+
+[<CliAnalyzer("UnnamedDiscriminatedUnionFieldAnalyzer",
+              "Verifies each field in a union case is named.",
+              "https://ionide.io/ionide-analyzers/hints/004.html")>]
+let unnamedDiscriminatedUnionFieldAnalyzer (ctx: CliContext) =
+    async {
+        let messages = ResizeArray<Message>()
+
+        let visitor =
+            { new SyntaxCollectorBase() with
+                override x.WalkUnionCase(unionCase: SynUnionCase) =
+                    match unionCase with
+                    | SynUnionCase(caseType = SynUnionCaseKind.Fields [ _ ]) -> ()
+                    | SynUnionCase(caseType = SynUnionCaseKind.Fields fields) ->
+                        let unNamedFields =
+                            fields
+                            |> List.choose (fun (SynField(idOpt = idOpt; range = mField)) ->
+                                if Option.isNone idOpt then Some mField else None
+                            )
+
+                        for mField in unNamedFields do
+                            messages.Add
+                                {
+                                    Type = "UnnamedDiscriminatedUnionFieldAnalyzer"
+                                    Message = "Field inside union case is not named!"
+                                    Code = "004"
+                                    Severity = Severity.Error
```

Comment:
`Severity.Error` feels a bit too strong to me as the default severity level.
But as there's no real right or wrong with these things, I'm also ok with keeping it at `Error`.

Reply:
Ah yes, that was actually temporary. I noticed that `Hint` (which I previously had) isn't getting picked up by the Sarif file. So I changed it to `Error`.
I'll revert it.